db/cache/charts.go

```diff
@@ -0,0 +1,1015 @@
+// it does exist and false if otherwise.
+func isfileExists(filePath string) bool {
+	_, err := os.Stat(filePath)
+	return !os.IsNotExist(err)
+}
+
+// writeCacheFile creates the charts cache in the provided file path if it
+// doesn't exists. It dumps the ChartsData contents using the .gob encoding.
+func (charts *ChartData) writeCacheFile(filePath string) (err error) {
+	var file *os.File
+	if !isfileExists(filePath) {
+		file, err = os.Create(filePath)
+	} else {
+		file, err = os.Open(filePath)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	defer file.Close()
+
+	encoder := gob.NewEncoder(file)
+	charts.mtx.RLock()
+	defer charts.mtx.RUnlock()
+	return encoder.Encode(charts.gobject())
+}
+
+// readCacheFile reads the contents of the charts cache dump file encoded in
+// .gob format if it exists returns an error if otherwise. It then deletes
+// the read *.gob cache dump file.
+func (charts *ChartData) readCacheFile(filePath string) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		file.Close()
+
+		// delete the dump after reading.
+		os.RemoveAll(filePath)
+	}()
+
+	var gobject = new(ChartGobject)
+
+	decoder := gob.NewDecoder(file)
+	err = decoder.Decode(&gobject)
+	if err != nil {
+		return err
+	}
+
+	charts.mtx.Lock()
+	charts.Blocks.Height = gobject.Height
+	charts.Blocks.Time = gobject.Time
+	charts.Blocks.PoolSize = gobject.PoolSize
+	charts.Blocks.PoolValue = gobject.PoolValue
+	charts.Blocks.BlockSize = gobject.BlockSize
+	charts.Blocks.TxCount = gobject.TxCount
+	charts.Blocks.NewAtoms = gobject.NewAtoms
+	charts.Blocks.Chainwork = gobject.Chainwork
+	charts.Blocks.Fees = gobject.Fees
+	charts.Windows.Time = gobject.WindowTime
+	charts.Windows.PowDiff = gobject.PowDiff
+	charts.Windows.TicketPrice = gobject.TicketPrice
+	charts.mtx.Unlock()
+
+	err = charts.Lengthen()
+	if err != nil {
+		log.Warnf("problem detected during (*ChartData).Lengthen. clearing datasets: %v", err)
+		charts.Blocks.Snip(0)
+		charts.Windows.Snip(0)
+		charts.Days.Snip(0)
+	}
+
+	return nil
+}
+
+// Load loads chart data from the gob file at the specified path and performs an
+// update.
+func (charts *ChartData) Load(cacheDumpPath string) {
+	t := time.Now()
+
+	if err := charts.readCacheFile(cacheDumpPath); err != nil {
+		log.Debugf("Cache dump data loading failed: %v", err)
+	}
+
+	// Bring the charts up to date.
+	charts.Update()
+
+	log.Debugf("Completed the initial chart load in %d s", time.Since(t).Seconds())
+}
+
+// Dump dumps a ChartGobject to a gob file at the given path.
+func (charts *ChartData) Dump(dumpPath string) {
+	err := charts.writeCacheFile(dumpPath)
+	if err != nil {
+		log.Errorf("ChartData.writeCacheFile failed: %v", err)
+	} else {
+		log.Debug("Dumping the charts cache data was successful")
+	}
+}
+
+// Store triggers an Update. Satisfies blockdata.BlockDataSaver.
+func (charts *ChartData) Store(_ *blockdata.BlockData, _ *wire.MsgBlock) error {
```

Comment:
The `blockdata` dependency is somewhat unfortunate, esp. since it does not actually need the data.  But it seems that other `BlockDataSavers` are already running `charts.Update`, so can this Store method be removed?

Reply:
I'm actually hoping to go the other way, removing charts from the explorer package altogether, since explorer doesn't use them. 

I also hate the unneeded dependency. I've got an idea for a solution.